front/src/modules/rollingStock/components/RollingStockSelector/RollingStockCurve.tsx

```diff
@@ -140,32 +172,45 @@ export default function RollingStockCurve({
   data,
   curvesComfortList,
   isOnEditionMode,
+  showPowerRestriction,
+  hoveredElectricalProfile,
 }: {
   data: EffortCurvesModes;
   curvesComfortList: Comfort[];
   isOnEditionMode?: boolean;
+  showPowerRestriction?: boolean;
+  hoveredElectricalProfile?: string | null;
 }) {
   const { t, ready } = useTranslation(['rollingstock']);
   const mode2name = (mode: string) => (mode !== 'thermal' ? `${mode}V` : t('thermal'));
+  const selectedElectricalProfile = useSelector(getElectricalProfile);
 
   const transformedData = useMemo(() => {
     const transformedCurves: TransformedCurves = {};
     Object.keys(data).forEach((mode) => {
-      // Standard curves (required)
       const name = mode2name(mode);
-      transformedCurves[`${name} STANDARD`] = {
-        ...(data[mode].default_curve as TransformedCurves['index']),
-        mode: name,
-        comfort: 'STANDARD',
-      };
-      // AC & HEATING curves (optional)
+      const isThermalMode = mode === THERMAL_TRACTION_IDENTIFIER;
       data[mode].curves.forEach((curve) => {
-        if (curve.cond?.comfort) {
-          const optionalCurveName = `${name} ${curve.cond.comfort}`;
-          transformedCurves[optionalCurveName] = {
+        if (
+          curve.cond?.comfort &&
+          (curve.cond?.electrical_profile_level || isNull(curve.cond?.electrical_profile_level))
+        ) {
+          const electrical_profil = isOnEditionMode
+            ? isParamNull(curve.cond.electrical_profile_level)
+            : '';
+          const power_restriction =
+            showPowerRestriction && curve.cond.power_restriction_code
+              ? isParamNull(curve.cond.power_restriction_code)
+              : '';
+          const curveName = `${name} ${curve.cond.comfort}${electrical_profil}${power_restriction}`;
+          transformedCurves[curveName] = {
             ...(curve.curve as TransformedCurves['index']),
```

Comment:
This can be removed once the types are fixed in editoast and rtk
```suggestion
            ...(curve.curve as as {
              max_efforts: number[];
              speeds: number[];
            }),
```